pybamm/expression_tree/symbol.py

```diff
@@ -354,13 +354,22 @@ def is_constant(self):
         evaluate : evaluate the expression
 
         """
-
         # if any of the nodes are instances of any of these types, then the whole
         # expression depends on either t or y
         search_types = (pybamm.Variable, pybamm.StateVector, pybamm.IndependentVariable)
 
         # do the search, return true if no relevent nodes are found
-        return all([not isinstance(n, search_types) for n in self.pre_order()])
+        # a Scalar with a name different to its is not constant, for parameter changing
```

Comment:
why wouldn't a Scalar with a name not be constant? I presume that the parameters wouldn't be changed *during* a simulation?